operator/src/main/java/oracle/kubernetes/operator/rest/WebhookRestServer.java

```diff
@@ -0,0 +1,319 @@
+   * if singleton instance not created.
+   */
+  public static void destroy() {
+    LOGGER.entering();
+    try {
+      if (INSTANCE != null) {
+        INSTANCE = null;
+        return;
+      }
+
+      throw new IllegalStateException();
+    } finally {
+      LOGGER.exiting();
+    }
+  }
+
+  /**
+   * Defines a resource configuration that scans for JAX-RS resources and providers in the REST
+   * package.
+   *
+   * @param restConfig the operator REST configuration
+   * @return a resource configuration
+   */
+  static ResourceConfig createResourceConfig(WebhookRestConfig restConfig) {
+    ResourceConfig rc =
+        new ResourceConfig()
+            .register(JacksonFeature.class)
+            .register(ErrorFilter.class)
+            .register(RequestDebugLoggingFilter.class)
+            .register(ResponseDebugLoggingFilter.class)
+            .register(ExceptionMapper.class)
+            .packages(ConversionWebhookResource.class.getPackageName());
+    rc.setProperties(Map.of(WebhookRestConfig.WEBHOOK_REST_CONFIG_PROPERTY, restConfig));
+    return rc;
+  }
+
+  private ResourceConfig createResourceConfig() {
+    LOGGER.entering();
+
+    ResourceConfig rc = createResourceConfig(config);
+
+    LOGGER.exiting();
+    return rc;
+  }
+
+  private static byte[] readFromDataOrFile(String data, String file) throws IOException {
+    if (data != null && data.length() > 0) {
+      return Base64.decodeBase64(data);
+    }
+    return Files.readAllBytes(new File(file).toPath());
+  }
+
+  /**
+   * Returns the in-pod URI of the available https REST port.
+   *
+   * @return the uri
+   */
+  String getHttpsUri() {
+    return baseHttpsUri;
+  }
+
+  /**
+   * Starts Webhook REST api for WebLogic operator.
+   *
+   * <p>If a port has not been configured, then it logs that fact, does not start that port, and
+   * continues (v.s. throwing an exception and not starting any ports).
+   *
+   * @param container Container
+   * @throws Exception if the REST api could not be started for reasons other than a port was not
+   *     configured. When an exception is thrown, then none of the ports will be leftrunning,
+   *     however it is still OK to call stop (which will be a no-op).
+   */
+  public void start(Container container) throws Exception {
+    LOGGER.entering();
+    if (httpsServer != null) {
+      throw new AssertionError("Already started");
+    }
+    boolean fullyStarted = false;
+    try {
+      httpsServer = createHttpsServer(container);
+      LOGGER.info(
+          "Started the Webhook REST server on "
+              + getHttpsUri()
+              + "/webhook"); // TBD .fine ?
+      fullyStarted = true;
+    } finally {
+      if (!fullyStarted) {
+        // if we didn't get a chance to start all of the ports because an exception
+        // was thrown, then stop the ones we did manage to start
+        stop();
+      }
+    }
+    LOGGER.exiting();
+  }
+
+  /**
+   * Stops WebLogic operator's REST api.
```

Comment:
minor,  it is stopping the webhook rest server not the operator's rest API?

Reply:
These should be two different REST servers.

Reply:
Fixed.